### PR TITLE
fix: Js editor inconsistency

### LINKED ISSUE #11008: [Bug]: Make JS Object page consistent with Query and Api pages.
Description

- The name/ title component of the JSObject is different from that of Apis and Queries.

Screenshot 2022-02-08 at 11 26 12 PM

Screenshot 2022-02-08 at 11 26 06 PM

- The JS Object UI doesn't seem the same as our queries. It is moving positions like in the video below

https://user-images.githubusercontent.com/46670083/166943948-d324f444-eebb-49e2-b9b2-2121727df4f9.mov

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

See Description

### Steps To Reproduce

1. Create a JSObject. See the UI. 
2. Create an API. See the UI.
3. Create a Query. See the UI

### Environment

Production

### Version

Cloud

### LINKED ISSUE #12303: [Bug]: JSObjects duplicate name popover is different in style as compared to other entities
### Is there an existing issue for this?

- [X] I have searched the existing issues

### **JSObjects: -.png**
Screenshot 2022-03-28 at 6 09 50 PM

### **API: -.png**
Screenshot 2022-03-28 at 6 07 02 PM

### **DataSources: - .png**

Screenshot 2022-03-28 at 6 08 08 PM

### Steps To Reproduce

1. Create new API/DataSources/JSObjects with name1 for example
2. Create another new API/DataSources with same name1 & observer the duplicate name pop over message style
3. Create new JSObjects with same name1 & observer the duplicate name pop over message style is different from other duplicate name pop over messages
4. **Expected: Style for message displayed should be in sync across all entities in app to maintain consistency**

### Public Sample App

_No response_

### Version

Cloud/Release v1.6.17

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

This PR makes the JS Editor page consistent with that of the Query page

Fixes #11008 
Fixes #12303

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manual (Only UI changes)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: js-editor-inconsistency 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.62 **(0.01)** | 38.54 **(-0.01)** | 35.91 **(0.01)** | 56.86 **(0)**
 :red_circle: | app/client/src/pages/Editor/JSEditor/Form.tsx | 28.77 **(-0.96)** | 0 **(0)** | 0 **(0)** | 32.81 **(-1.04)**
 :red_circle: | app/client/src/pages/Editor/JSEditor/JSObjectNameEditor.tsx | 53.13 **(-0.44)** | 20 **(0)** | 0 **(0)** | 53.13 **(-0.44)**
 :green_circle: | app/client/src/pages/Editor/JSEditor/styledComponents.ts | 50 **(7.14)** | 88.89 **(0)** | 0 **(0)** | 52.17 **(7.73)**</details>

### CODE REVIEW

**Review by rimildeyjsr — APPROVED** [review]
LGTM

File: app/client/src/pages/Editor/JSEditor/JSObjectNameEditor.tsx [hunk 1]
```diff
@@ -84,27 +95,35 @@ export function JSObjectNameEditor(props: ActionNameEditorProps) {
...
-            savingState={
-              saveStatus.isSaving
-                ? SavingState.STARTED
-                : SavingState.NOT_STARTED
-            }
-            underline
-            valueTransform={removeSpecialChars}
-          />
+          <div
+            style={{
```
Comment (Rishabh-Rathod) [inline app/client/src/pages/Editor/JSEditor/JSObjectNameEditor.tsx:99]: nit-pick:- Can we avoid inline style?

Comment (ohansFavour) [reply app/client/src/pages/Editor/JSEditor/JSObjectNameEditor.tsx]: done

**Comment by Rishabh-Rathod** [comment]
Won't be able to review this PR soon. Removing myself from the reviewer list for now.

**Review by rimildeyjsr — APPROVED** [review]

**Comment by Aishwarya-U-R** [comment]
@ohansFavour - minor feedback.. while we click on the JSObj name to edit it, can we give it a bit more empty space like we have it for Api's & Queries today. Because currently it gives me a feeling that there is not much space to rename even though the field increased in size after I start typing. But the first look of it does'nt convey the same:

Screenshot 2022-05-18 at 1 51 22 AM

API & Query:
Screenshot 2022-05-18 at 1 54 41 AM
Screenshot 2022-05-18 at 1 55 10 AM

**Pull request merged**